src/main.cpp

```diff
@@ -744,34 +799,30 @@ int main(int argc, char* argv[])
     }
     else
     {
-      std::ifstream file(filename);
+      std::ifstream file(args.filename);
       if (file.fail())
       {
-        LOG(ERROR) << "failed to open file '" << filename
+        LOG(ERROR) << "failed to open file '" << args.filename
                    << "': " << std::strerror(errno);
         return -1;
       }
 
+      filename = args.filename;
```

Comment:
I believe that `filename` could be completely removed in favor of `args.filename`.

Reply:
I debated this too. So far, I've taken the view that the parsed args should be constant:

```
const Args args = parse_args(argc, argv);
```

The rationale is to try to nudge manipulations of arguments into the `parse_args()` function, rather than doing them in the body. Of course, this turns out to be a counter-example because it ends up causing the creation of another variable.

Anyways, I'm not rigid on this. If the preference is still to switch this, I'm happy to do so.

Reply:
Me neither, I think that we can leave it is.